src/main/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4Kit.kt

```diff
@@ -439,11 +439,21 @@ class GoogleAnalyticsFirebaseGA4Kit : KitIntegration(), KitIntegration.EventList
         products?.let {
             val bundles = arrayOfNulls<Bundle>(products.size)
             for ((i, product) in products.withIndex()) {
-                val bundle = getBundle(product)
-                bundles[i] = bundle.bundle
+                val bundle = getHashmap(product)
+                val bundleSize = bundle.size
+                if (bundleSize > itemMaxParameter) {
+                    val additionalParameterSize = bundleSize - itemMaxParameter
```

Comment:
Shouldn't this be reversed?  
`itemMaxParameter - bundleSize`?

Right now the the bundle would have up to 4 parameters, (quantity, itemId, itemName, and category), and the itemMaxParameter is 25.  So wouldn't bundleSize - itemMaxParameter be negative?

Reply:
So, the subsequent line, if (bundleSize > itemMaxParameter), will only be executed when the bundle size exceeds the itemMaxParameter. For example, if the bundle size is 4, line 445 will not be executed.

Reply:
Resolving - I understand that this is just reaching feature parity with iOS, which is also missing adding product parameters to the bundle